### PR TITLE
Add tags for gdal_edit processing algorithm

### PR DESCRIPTION
**Pull request by AlisterH**
## Description

Add tags for gdal_edit processing algorithm (GDAL>Raster projections>Assign projection).
Now it is consistent with the vector "assign projection algorithm", and I can find it by searching e.g. for crs.

## Note

Tagging of processing algorithms is currently pretty inconsistent and could do with a systematic review.
It particularly bothers me when a search will find a vector algorithm but not a raster algorithm that does the same thing.

### CODE REVIEW

File: python/plugins/processing/algs/gdal/AssignProjection.py [hunk 1]
```diff
@@ -64,6 +64,9 @@ def displayName(self):
     def icon(self):
         return QIcon(os.path.join(pluginPath, 'images', 'gdaltools', 'projection-add.png'))
 
+    def tags(self):
+        return ['assign', 'set', 'transform', 'reproject', 'crs', 'srs', 'warp', self.commandName()]
```
Comment (alexbruy) [inline python/plugins/processing/algs/gdal/AssignProjection.py:68]:
```suggestion
        tags = self.tr('assign,set,transform,reproject,crs,srs').split(',')
        tags.extend(super().tags())
        return tags
```

**Review by alexbruy — APPROVED** [review]

**Comment by AlisterH** [comment]
If I ever find the time I was thinking of going through all the stock algorithms to review the tagging.
@alexbruy is there any chance you could explain the purpose of your change?  I presume you'd want the same change done for other algorithms that don't already do it that way.

**Comment by nyalldawson** [comment]
> is there any chance you could explain the purpose of your change

There's two parts to it:
- using "tr" with a comma delimited list which is then split makes the tags translation friendly. Translators can then translate the whole string to a comma delimited list in the target language, adding or removing tags which don't make sense in that language.
- adding the base class' tags means that tags added to the base class will always be returned. In the case of the gdal providers there's logic in the base class to automatically create some tags using the gdal command name.

**Comment by alexbruy** [comment]
Oh, I see @nyalldawson already answered. Basically it makes tags useful for non-English users by making them translatable and preserves tags defined in the base class.

**Pull request merged**